strax/processing/general.py

```diff
@@ -4,26 +4,39 @@
 import numba
 from numba.typed import List
 import numpy as np
+import warnings
 
 export, __all__ = strax.exporter()
 
 
-# (5-10x) faster than np.sort(order=...), as np.sort looks at all fields
-# TODO: maybe this should be a factory?
 @export
-@numba.jit(nopython=True, nogil=True, cache=True)
 def sort_by_time(x):
     """Sort pulses by time, then channel.
-
-    Assumes you have no more than 10k channels, and records don't span
-    more than 100 days. TODO: FIX this
     """
     if len(x) == 0:
-        # Nothing to do, and .min() on empty array doesn't work, so:
         return x
+    with warnings.catch_warnings():
+        warnings.filterwarnings('error')
+        try:
+            max_channeL_plus_one = x['channel'].max() + 1
+            # Check if we trigger an overflow warning:
+            (x['time'].max() - x['time'].min()) * max_channeL_plus_one
+            x = _sort_by_time(x, max_channeL_plus_one)
+        except RuntimeWarning:
```

Comment:
this is rather crude, we know when we are going to have an overflow, you can easily calculate it, why not check that instead of trying and thereby doubling the work?

Reply:
Hmm, we could also do something like 
```
if not (x['time'].max() - x['time'].min())  > np.iinfo(np.int64).max / max_channel_plus_one
```
But I was a bit afraid of the division.

Reply:
right, but you can have an off by a few and make the switch earlier - that wouldn't matter in the end

By the way, any idea why the tests in straxen fail - it seems like something not working?

Reply:
No not yet, I saw it but was not yet in the mood to check why :D